6.recursiveSum.js

```diff
@@ -8,10 +8,7 @@
 // recursiveSum(arr1) should return 27
 
 
-const recursiveSum = (arr) => {
-
-}
-
-
+const recursiveSum = (arr) => !arr.length ? 0 : arr[0] + recursiveSum(arr.slice(1))
+//const recursiveSum = (arr) => !arr.length ? 0 : arr.pop() + recursiveSum(arr)
```

Comment:
By doing slice each time I think you may be making this problem n^2. Instead add an i in the argument.

Reply:
Oh yes, I thought of that, but wasn't sure if it would mess up the specs. Damn should have tried. Good recommend.